### PR TITLE
[IMP] mail: introduce suggestion list item models

### PR DESCRIPTION
**Pull request by seb-odoo**
This allows removing the AND/OR mix in identifying fields of ComposerSuggestionView.

Part of task-2741982

### CODE REVIEW

File: addons/mail/static/src/models/composer_view.js [hunk 1]
```diff
@@ -1572,7 +1572,7 @@ registerModel({
          * Determines whether the currently active suggestion should be scrolled
          * into view.
          */
-        hasToScrollToActiveSuggestion: attr({
+        hasToScrollToactiveSuggestionView: attr({
```
Comment (oomsveta) [inline addons/mail/static/src/models/composer_view.js:1575]: bad capitalization

**Review by oomsveta — CHANGES_REQUESTED** [review]

File: addons/mail/static/src/models/composer_view.js [hunk 2]
```diff
@@ -1593,13 +1478,8 @@ registerModel({
         textareaLastInputValue: attr({
```
Comment (oomsveta) [inline addons/mail/static/src/models/composer_view.js:1478]: While you are at it, maybe you could reorder this?

File: addons/mail/static/src/models/composer_view.js [hunk 3]
```diff
@@ -958,7 +873,7 @@ registerModel({
          * @return {boolean}
          */
         _computeHasSuggestions() {
-            return this.mainSuggestions.length > 0 || this.extraSuggestions.length > 0;
+            return this.mainSuggestables.length > 0 || this.extraSuggestables.length > 0;
```
Comment (oomsveta) [inline addons/mail/static/src/models/composer_view.js:876]: from what the compute does, shouldn't the field be called `hasSuggestables`?

File: addons/mail/static/src/models/composer_view.js [hunk 4]
```diff
@@ -445,9 +436,9 @@ registerModel({
                         this.threadView.startEditingLastMessageFromCurrentUser();
                         break;
                     }
-                    if (this.hasSuggestions) {
-                        this.setPreviousSuggestionActive();
-                        this.update({ hasToScrollToActiveSuggestion: true });
+                    if (this.composerSuggestionListView) {
```
Comment (oomsveta) [inline addons/mail/static/src/models/composer_view.js:439]: what prevents you from replacing `hasSuggestions` with `composerSuggestionListView` everywhere? and getting rid of `hasSuggestions` in the process?

File: addons/mail/static/src/models/composer_view.js (hunk 2 above)
Comment (seb-odoo) [reply addons/mail/static/src/models/composer_view.js:1478]: no useless diff

File: addons/mail/static/src/models/composer_view.js (hunk 4 above)
Comment (seb-odoo) [reply addons/mail/static/src/models/composer_view.js:439]: `hasSuggestions` is a data concept, `composerSuggestionListView` is a view concept, while there are overlaps (example in this case where I made the diff) it's not strictly always the case in theory.

All this code requires further clean up, I must made what I felt necessary to fix the identifying fields situation.

File: addons/mail/static/src/models/composer_view.js (hunk 3 above)
Comment (seb-odoo) [reply addons/mail/static/src/models/composer_view.js]: I renamed the lists to keep "suggestion" in them.

**Comment by oomsveta** [comment]
red runbot

**Comment by seb-odoo** [comment]
green runbot

**Review by oomsveta — APPROVED** [review]
@robodoo r+

**Comment by oomsveta** [comment]
@robodoo rebase-ff

**Pull request closed without merging**